app/features/signup/verify-email-form.tsx

```diff
@@ -82,10 +93,32 @@ export function VerifyEmailForm() {
     }
   };
 
-  if (user.email_verified || isGuestUser) {
+  useEffect(() => {
+    if (code) {
+      const handleVerificationCode = async () => {
+        const verified = await submitRef.current({ code });
+        if (!verified) {
+          toast({
+            variant: 'destructive',
```

Comment:
will this end up showing two toasts? It looks like if email verification fails in `onSubmit`, then there is a toast there too

Reply:
good catch, it will. will also be fixed in new version

Reply:
should be fixed now